nav2_behaviors/plugins/assisted_teleop.cpp

```diff
@@ -0,0 +1,167 @@
+
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  vel_sub_ = node->create_subscription<geometry_msgs::msg::Twist>(
+    input_vel_topic_, rclcpp::SystemDefaultsQoS(),
+    std::bind(
+      &AssistedTeleop::inputVelocityCallback,
+      this, std::placeholders::_1));
+
+  joy_sub_ = node->create_subscription<sensor_msgs::msg::Joy>(
+    joystick_topic_, rclcpp::SystemDefaultsQoS(),
+    std::bind(
+      &AssistedTeleop::joyCallback,
+      this, std::placeholders::_1));
+
+  return Status::SUCCEEDED;
+}
+
+void AssistedTeleop::onActionCompletion()
+{
+  RCLCPP_INFO(logger_, "Action completed");
+  input_twist_ = geometry_msgs::msg::Twist();
+  vel_sub_.reset();
+  joy_sub_.reset();
+}
+
+Status AssistedTeleop::onCycleUpdate()
+{
+  rclcpp::Duration time_remaining = end_time_ - steady_clock_.now();
+  if (time_remaining.seconds() < 0.0 && command_time_allowance_.seconds() > 0.0) {
+    stopRobot();
+    RCLCPP_WARN(
+      logger_,
+      "Exceeded time allowance before reaching the BackUp goal - Exiting BackUp");
+    return Status::FAILED;
+  }
+
+  geometry_msgs::msg::PoseStamped current_pose;
+  if (!nav2_util::getCurrentPose(
+      current_pose, *tf_, global_frame_, robot_base_frame_,
+      transform_tolerance_))
+  {
+    RCLCPP_ERROR(logger_, "Current robot pose is not available.");
+    return Status::FAILED;
+  }
+
+  // user states that teleop was successful
+  if (joy_.buttons.size() > 0 && joy_.buttons[0] > 0) {
+    stopRobot();
+    return Status::SUCCEEDED;
+  }
+
+  // user states that teleop failed
+  if (joy_.buttons.size() > 1 && joy_.buttons[1] > 0) {
+    stopRobot();
+    return Status::FAILED;
+  }
+
+  geometry_msgs::msg::Twist cmd_vel;
+  cmd_vel = computeVelocity(input_twist_);
+  vel_pub_->publish(std::move(cmd_vel));
+
+  return Status::RUNNING;
+}
+
+geometry_msgs::msg::Twist AssistedTeleop::computeVelocity(geometry_msgs::msg::Twist & twist)
+{
+  // TODO(jwallace42): Waiting for safety node to be completed.
+  return twist;
+}
+
+geometry_msgs::msg::Pose2D AssistedTeleop::projectPose(
+  geometry_msgs::msg::Pose2D pose,
+  geometry_msgs::msg::Twist twist,
+  double projection_time)
+{
+  geometry_msgs::msg::Pose2D projected_pose;
+
+  projected_pose.x = projection_time * (
+    twist.linear.x * cos(pose.theta) +
+    twist.linear.y * sin(pose.theta));
+
+  projected_pose.y = projection_time * (
+    twist.linear.x * sin(pose.theta) -
+    twist.linear.y * cos(pose.theta));
+
+  projected_pose.theta = projection_time * twist.angular.z;
+
+  return projected_pose;
+}
+
+void AssistedTeleop::inputVelocityCallback(const geometry_msgs::msg::Twist::SharedPtr msg)
+{
+  RCLCPP_INFO(logger_, "Hit input Vel callback");
+  input_twist_.linear.x = msg->linear.x;
+  input_twist_.linear.y = msg->linear.y;
+  input_twist_.angular.z = msg->angular.z;
+}
+
+void AssistedTeleop::joyCallback(const sensor_msgs::msg::Joy msg)
+{
+  RCLCPP_INFO(logger_, "Hit joy callback");
```

Comment:
Remove logging

Also, should the joy stick be removed? I don't think there is a standard for how joystick layouts should be processed for velocity commands. Something else needs to translate joystick->velocity for us so we know what pressing "A" means